app/core/management/utils/xia_internal.py

```
@@ -0,0 +1,230 @@
+                                required_column_list)
+
+        else:
+            update_flattened_object(list_obj[i], prefix, flatten_dict)
+
+        # looping through required column names
+        for required_prefix in required_column_list:
+            # finding matching value along with index
+            try:
+                required_prefix.index(prefix)
+            except ValueError:
+                continue
+            else:
+                if required_prefix.index(prefix) == 0:
+                    required_prefix_list.append(required_prefix)
+        #  setting up flag for checking validation
+        passed = True
+
+        # looping through items in required columns with matching prefix
+        for item_to_check in required_prefix_list:
+            #  flag if value not found
+            if not flatten_dict[item_to_check]:
+                passed = False
+
+        # if all required values are skip other object in list
+        if passed:
+            break
+
+
+def flatten_dict_object(dict_obj, prefix, flatten_dict, required_column_list):
+    """function to flatten dictionary object"""
+    for element in dict_obj:
+        if isinstance(dict_obj[element], dict):
+            flatten_dict_object(dict_obj[element], prefix + "." +
+                                element, flatten_dict, required_column_list)
+
+        elif isinstance(dict_obj[element], list):
+            flatten_list_object(dict_obj[element], prefix + "." +
+                                element, flatten_dict, required_column_list)
+
+        else:
+            update_flattened_object(dict_obj[element], prefix + "." +
+                                    element, flatten_dict)
+
+
+def update_flattened_object(str_obj, prefix, flatten_dict):
+    """function to update flattened object to dict variable"""
+
+    flatten_dict.update({prefix: str_obj})
+
+
+def myconverter(o):
+    if isinstance(o, datetime.datetime):
+        o = o.isoformat()
+        return o
```

Also if the argument value is not an instance of `date.datetime` then this function will implicitly return null -- i'm not sure if that's what we want.  It might be better to return the original value?  If so this can be handled by simply outdenting the `return` statemetn

---

Or maybe we want to raise a conversion exception that could be caught by the caller.